packages/webapp/src/containers/Animals/Inventory/useAnimalOrBatchRemoval.ts

```diff
@@ -13,29 +13,43 @@
  *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
  */
 
-import { Dispatch, SetStateAction, useState } from 'react';
+import { Dispatch, SetStateAction, useState, useEffect } from 'react';
 import {
+  useDeleteAnimalBatchesMutation,
+  useDeleteAnimalsMutation,
   useRemoveAnimalBatchesMutation,
   useRemoveAnimalsMutation,
 } from '../../../store/api/apiSlice';
 import { toLocalISOString } from '../../../util/moment';
 import { parseInventoryId } from '../../../util/animal';
-import { FormFields } from '../../../components/Animals/RemoveAnimalsModal';
+import { CREATED_IN_ERROR_ID, FormFields } from '../../../components/Animals/RemoveAnimalsModal';
+import useMutations from '../../../hooks/api/useMutations';
+import { enqueueErrorSnackbar, enqueueSuccessSnackbar } from '../../Snackbar/snackbarSlice';
+import { TFunction } from 'i18next';
 import { AnimalOrBatchKeys } from '../types';
 
 const useAnimalOrBatchRemoval = (
   selectedInventoryIds: string[],
   setSelectedInventoryIds: Dispatch<SetStateAction<string[]>>,
+  dispatch: Dispatch<any>,
+  t: TFunction,
 ) => {
-  const [removeAnimals] = useRemoveAnimalsMutation();
-  const [removeBatches] = useRemoveAnimalBatchesMutation();
+  const { mutations } = useMutations([
```

Comment:
We sometimes need to combine the results of hooks for GET APIs using `useQueries`, but I don't see any benefit in combining mutations here... do we want to wait for two APIs to return the results and show snackbars at the same time?

Reply:
There might be some value in the isSuccess (all succeed) and isLoading (still waiting) pieces... since I think those can be genericized. 

But the main usefulness I think is renaming 😂 . Having to rename all the isXXXLoading, isXXXError gets cumbersome in the same component. So having one object hold them all with a named label is useful to me.

Reply:
And I thiiiink we want to show them at the same time, it might be weird if they are staggered. 

It sounds like we are going to put effort towards the partial success in the near future so this is temporary

Reply:
Renaming - true, I've experienced the same thing 😅 

I don't mind subtle difference in timing, but I think all the `dispatch(...Snackbar())` should be written after those two try-catch blocks.

Oh it's temporary so you don't mind it too 😂